libpromises/processes_select.c

```diff
@@ -548,6 +548,29 @@ static bool SelectProcRegexMatch(const char *name1, const char *name2,
     return false;
 }
 
+static void PrintStringIndexLine(int prefix_spaces, int len)
+{
+    char arrow_str[CF_BUFSIZE];
+    arrow_str[0] = '^';
+    arrow_str[1] = '\0';
+    char index_str[CF_BUFSIZE];
+    index_str[0] = '0';
+    index_str[1] = '\0';
+    for (int lineindex = 10; lineindex <= len; lineindex += 10)
+    {
+        char num[11];
+        xsnprintf(num, sizeof(num), "% 10d", lineindex);
```

Comment:
No, that's correct! :-) The complete expression means: "print a number, make it ten characters wide, and fill the unused space with the space character".